computation-slurm/src/main/java/com/powsybl/computation/slurm/SlurmComputationManager.java

```diff
@@ -312,9 +312,10 @@ private void scancel(Long jobId) {
 
             int sum = commandExecutions.stream().mapToInt(CommandExecution::getExecutionCount).sum();
             TaskCounter taskCounter = new TaskCounter(sum);
+            taskStore.insert(remoteWorkingDir.getFileName().toString(), taskCounter);
```

Comment:
@yichen88 : I think there is still a "hole" after this fix.
It may happen that the client calls f.cancel() before the executor has started to execute the task.
In this case, the call to cancel() will wait for the thread to be set (which is "weird" because it may block for a while, but it's not the topic of that issue), and try to retrieve the taskCounter right after. But at this point, it may not have been set.

I think we should either :
 * ensure that the `taskCounter` is set before this potential call to `cancel`, for example before returning the future
 * ensure that the `cancel` method can handle a null `taskCounter`

Also, maybe we should make an effort to have unit test (not integration tests) to test those mechanics. Maybe we should try to have an inner interface with Slurm that we could mock?